### PR TITLE
Fix EC2 AZ Subnet Determination for VPC

### PR DESCRIPTION
**Pull request by manadart**
### Checklist

 - [x] Checked if it requires a [pylibjuju](https://github.com/juju/python-libjuju) change?
 - [x] Added [integration tests](https://github.com/juju/juju/tree/develop/tests) for the PR?
 - [x] Added or updated [doc.go](https://discourse.jujucharms.com/t/readme-in-packages/451) related to packages changed?
 - [x] Do comments answer the question of why design decisions were made?

----

## Description of change

Fixes an issue with EC2 where a VPC is set and the list of allowable subnet IDs is incorrectly passed when determining VPC subnets for the desired availability zone.

This bug was introduced by https://github.com/juju/juju/pull/11194 and turned up by integration tests.

## QA steps

```sh
./assess --juju ~/go/bin/juju --substrate aws --region eu-west-2 --logging-config '<root>=DEBUG' network_spaces
```

## Documentation changes

None.

## Bug reference

N/A

### CODE REVIEW

File: provider/ec2/environ.go [hunk 1]
```diff
@@ -565,6 +565,17 @@ func (e *environ) StartInstance(ctx context.ProviderCallContext, args environs.S
 	runArgs := commonRunArgs
 	runArgs.AvailZone = availabilityZone
 
+	// TODO (manadart 2020-02-07): We only take the first subnet/zones
+	// mapping to create a NIC for the instance.
+	// This effectively uses a single positive space constraint if many are
+	// specified; behaviour that dates from the original spaces MVP.
+	// It will not take too much effort to enable multi-NIC support for EC2
+	// if we use them all when constructing the instance creation request.
+	var subnetZones map[corenetwork.Id][]string
```
Comment (SimonRichardson) [inline provider/ec2/environ.go]: Just a note: if anybody does attempt to use this for filling, it will go down panic town...

Comment (manadart) [reply provider/ec2/environ.go]: True, but it is only ever read.

**Comment by manadart** [comment]
!!build!!

**Comment by manadart** [comment]
$$merge$$

**Pull request merged**